### PR TITLE
Security enhances

### PR DESCRIPTION
**Pull request by petermcd**
Added security enhancements

## Summary by Sourcery

Add security-focused validations and internal tooling updates for the lexer, along with versioning and test coverage adjustments.

New Features:
- Introduce a maximum allowed source size for the lexer to prevent processing excessively large inputs.
- Add a dedicated Security Lead agent configuration and command wiring for internal security reviews.

Bug Fixes:
- Tighten fail-state parsing and validation to reject out-of-range error codes and improve robustness of the fail-state regex.
- Adjust the assignment regex to correctly capture values without interfering with other token patterns.

Enhancements:
- Refactor lexer token specifications to store direct processor callables instead of method names, simplifying token processing and reducing indirection.
- Emit a structured warning log when unrecognised content is encountered and converted into a parse error token.

Build:
- Bump project version from 0.3.1 to 0.4.0 to reflect the new security-related changes.

Tests:
- Add tests covering the new maximum source size constraint and validation of fail-state error code ranges.

Chores:
- Add Claude security lead agent configuration and helper command files for standardized security assessments.

### CODE REVIEW

File: tpc_plugin_parser/lexer/utilities/regex.py [hunk 1]
```diff
@@ -1,17 +1,15 @@
 """List of regex the lexer uses."""
 
-ASSIGNMENT: str = (
-    r"^(?:[\s]*)(?P<name>[\w]+)(?:(?:[\s]*)(?P<equals>=)(?:(?:[\s]*)(?P<value>(?:(?!\s*fail\s*\().)+))?)?(?:[\s]*)$"
-)
+ASSIGNMENT: str = r"^(?:[\s]*)(?P<name>[\w]+)(?:(?:[\s]*)(?P<equals>=)(?:(?:[\s]*)(?P<value>[^\r\n]*))?)?(?:[\s]*)$"
```
[the remark this answers is not in the document]
Comment (petermcd) [reply tpc_plugin_parser/lexer/utilities/regex.py:3]: Fine

File: tpc_plugin_parser/lexer/lexer.py [hunk 2]
```diff
@@ -87,7 +94,7 @@ def process(self) -> None:
         for line_number, line in enumerate(self._source.splitlines(), start=1):
             for token_spec in self._token_specs:
                 if match := token_spec["pattern"].match(line):
-                    getattr(self, token_spec["processor_method"])(match=match, line_number=line_number)
+                    token_spec["processor"](match=match, line_number=line_number)
```
[the remark this answers is not in the document]
Comment (petermcd) [reply tpc_plugin_parser/lexer/lexer.py:97]: Ty didn't fail

File: tpc_plugin_parser/lexer/lexer.py [hunk 3]
```diff
@@ -43,35 +50,35 @@ def __init__(self, source: str) -> None:
...
-                "pattern": re.compile(ASSIGNMENT, re.IGNORECASE),
-                "token_name": TokenName.ASSIGNMENT,
-                "processor_method": "_process_assignment",
-            },
             {
                 "pattern": re.compile(COMMENT, re.IGNORECASE),
                 "token_name": TokenName.COMMENT,
-                "processor_method": "_process_comment",
+                "processor": self._process_comment,
             },
```
[the remark this answers is not in the document]
Comment (petermcd) [reply tpc_plugin_parser/lexer/lexer.py:57]: lever is only created once

File: tpc_plugin_parser/lexer/utilities/regex.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (petermcd) [reply tpc_plugin_parser/lexer/utilities/regex.py:3]: Testing passes happy with outcome

File: tpc_plugin_parser/lexer/lexer.py [hunk 4]
```diff
@@ -35,6 +39,9 @@ class Lexer(object):
     def __init__(self, source: str) -> None:
         """Standard init for the Lexer object."""
 
+        if len(source.encode()) > _MAX_SOURCE_BYTES:
+            raise ValueError(f"Source exceeds the maximum allowed size of {_MAX_SOURCE_BYTES} bytes")
```
[the remark this answers is not in the document]
Comment (petermcd) [reply tpc_plugin_parser/lexer/lexer.py:43]: edge case

**Comment by petermcd** [comment]
unable to merge GitHub no updating from settings
